ome/provider.go

```diff
@@ -129,105 +135,96 @@ func (p *provider) Configure(ctx context.Context, req tfsdk.ConfigureProviderReq
-	return map[string]tfsdk.ResourceType{
-		"ome_template":                 resourceTemplateType{},
-		"ome_deployment":               resourceDeploymentType{},
-		"ome_configuration_baseline":   resourceConfigurationBaselineType{},
-		"ome_configuration_compliance": resourceConfigurationComplianceType{},
-	}, nil
+func (p *omeProvider) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		NewTemplateResource,
+		NewDeploymentResource,
+		NewConfigurationBaselineResource,
+		NewConfigurationComplianceResource,
+	}
 }
 
-func (p *provider) GetDataSources(ctx context.Context) (map[string]tfsdk.DataSourceType, diag.Diagnostics) {
-	return map[string]tfsdk.DataSourceType{
-		"ome_template_info":             templateDataSourceType{},
-		"ome_groupdevices_info":         groupDevicesDataSourceType{},
-		"ome_vlannetworks_info":         vlanNetowrksDataSourceType{},
-		"ome_configuration_report_info": configurationReportDataSourceType{},
-	}, nil
+func (p *omeProvider) DataSources(ctx context.Context) []func() datasource.DataSource {
+	return []func() datasource.DataSource{
+		NewTemplateDataSource,
+		NewGroupDevicesDatasource,
+		NewVlanNetworkDataSource,
+		NewConfigurationReportDataSource,
+	}
 }
 
-func (p *provider) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
-	return tfsdk.Schema{
+func (p *omeProvider) Schema(ctx context.Context, _ provider.SchemaRequest, resp *provider.SchemaResponse) {
+	resp.Schema = schema.Schema{
 		MarkdownDescription: "The Terraform Provider for OpenManage Enterprise (OME) is a plugin for Terraform that allows the resource management of PowerEdge servers using OME",
-		Attributes: map[string]tfsdk.Attribute{
-			"host": {
+		Attributes: map[string]schema.Attribute{
+			"host": schema.StringAttribute{
 				MarkdownDescription: "OpenManage Enterprise IP address or hostname.",
 				Description:         "OpenManage Enterprise IP address or hostname.",
-				Type:                types.StringType,
 				Required:            true,
 			},
-			"username": {
+			"username": schema.StringAttribute{
 				MarkdownDescription: "OpenManage Enterprise username.",
 				Description:         "OpenManage Enterprise username.",
-				Type:                types.StringType,
 				Required:            true,
 			},
-			"password": {
+			"password": schema.StringAttribute{
 				MarkdownDescription: "OpenManage Enterprise password.",
 				Description:         "OpenManage Enterprise password.",
-				Type:                types.StringType,
 				Required:            true,
 				Sensitive:           true,
 			},
-			"port": {
+			"port": schema.Int64Attribute{
 				MarkdownDescription: "OpenManage Enterprise HTTPS port.",
 				Description:         "OpenManage Enterprise HTTPS port.",
-				Type:                types.Int64Type,
 				Optional:            true,
-				PlanModifiers: tfsdk.AttributePlanModifiers{
```

Comment:
I guess here we're skipping the default values.

Reply:
we don't have plan modifier in provider schema.
we have set the default value of port to 443 in provider configuration.